internal/services/dns/parse/soa_record.go

```diff
@@ -60,7 +60,7 @@ func SoaRecordID(input string) (*SoaRecordId, error) {
 		return nil, fmt.Errorf("ID was missing the 'resourceGroups' element")
 	}
 
-	if resourceId.DnszoneName, err = id.PopSegment("dnszones"); err != nil {
+	if resourceId.DnszoneName, err = id.PopSegment("dnsZones"); err != nil {
```

Comment:
This file is generated so unfortunately these changes will always be overwritten when the provider is built.